indico/web/client/js/react/components/DatePicker.jsx

```diff
@@ -0,0 +1,90 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import {Param, Translate} from 'indico/react/i18n';
+
+import 'indico/custom_elements/ind_date_picker';
+
+export default function DatePicker({
+  onChange,
+  value,
+  format = moment.localeData().longDateFormat('L'),
+  ...inputProps
+}) {
+  function handleDateChange(ev) {
+    const {date} = ev.target.closest('ind-date-picker');
+    // en-CA uses the ISO format date (YYYY-MM-DD) but gives it in local time zone.
+    // Do not use toISOString() for this because it may result in incorrect date due
+    // to time zone differences.
+    onChange(date?.toLocaleDateString('en-CA'));
+  }
+
+  return (
+    <ind-date-picker>
+      <input type="text" onChange={handleDateChange} defaultValue={value} {...inputProps} />
+      <button type="button">
+        <Translate as="span">Open a calendar</Translate>
+      </button>
+      <ind-calendar>
+        <dialog>
+          <div className="controls">
+            <button type="button" value="previous-year">
+              <Translate as="span">Previous year</Translate>
+            </button>
+            <button type="button" value="previous-month">
+              <Translate as="span">Previous month</Translate>
+            </button>
+
+            <div className="month-year">
+              <select />
+              <input type="number" min="1900" max="2100" required />
```

Comment:
I noticed that you can use the calendar to exceed this. For now I think this can be safely ignored, but when adding limitations on the earliest/latest date that can be selected, it should probably be taken into account (and maybe dynamically update the min/max of the year and month pickers here).

Reply:
Hm, yeah. I'll give it some thought tomorrow.

Reply:
I think the other open threads are higher priority, especially since the base regform field does not need limiting the min/max range anyway (even though the UN "variant" of it, which IIRC is used for birth dates / passport validity dates, might require this). I just left the comment since it's easy to forget such things later ;)

Reply:
I've decided to just remove the constraints for now. When I start working on the dual-field date range picker, I'll address the min-max constraint anyway. So for now I think this is good to go.